styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java

```diff
@@ -112,6 +121,19 @@
   public static final String PROPERTY_CONFIG_DEBUG_ENABLED = "debug";
   public static final String PROPERTY_SUBMISSION_RATE_LIMIT = "submissionRateLimit";
 
+  public static final String PROPERTY_STATE = "state";
+  public static final String PROPERTY_STATE_TIMESTAMP = "stateTimestamp";
```

Comment:
Seems a bit unexpected to have this list of constants mirroring StateData members but living outside of StateData.java.

Checked for analogues of Jackson annotations for Datastore, and found [@PersistenceCapable](https://cloud.google.com/appengine/docs/standard/java/datastore/jdo/dataclasses#Class_and_Field_Annotations)  - something worth exploring or already explored?

Reply:
I agree that it's a bit annoying code but I don't think it's a big deal. And I think the decoupling of datastore entity representation from the RunState/StateData layout (StateData in particular is exposed through the client api and is used in the cli) is a good thing.

Using JDO doesn't seem recommended, it's listed under `Superseded Storage Solutions` in the app engine docs. The cloud datastore docs do not seem to mention JDO at all. 

[Objectify](https://github.com/objectify/objectify) seems like a recommended alternative, but is apparently [app engine only](https://github.com/objectify/objectify/wiki/FrequentlyAskedQuestions#can-i-use-objectify-from-a-gae-flexible-runtime). I guess the same might be true of JDO.